group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntime.java

```diff
@@ -541,6 +541,28 @@ private void unload() {
         }
     }
 
+    class OperationTimeout extends TimerTask {
+        private final TopicPartition tp;
+        private final DeferredEvent event;
+
+        public OperationTimeout(
+            TopicPartition tp,
+            DeferredEvent event,
+            long delayMs
+        ) {
+            super(delayMs);
+            this.event = event;
+            this.tp = tp;
+        }
+
+        @Override
+        public void run() {
+            String name = event.toString();
+            scheduleInternalOperation("OperationTimeout", tp,
```

Comment:
previous operation name carries more information. Should we keep it? for instance: `"OperationTimeout(name=" + name + ", tp=" + tp + ")"`

Reply:
Let me add it back.

Reply:
Done.